add libfmt-dev to package.xml for ROS-O

k-okada opened the pull request:
this will fix 
```
2024-12-12T02:44:00.5472061Z dpkg-buildpackage
2024-12-12T02:44:00.5472253Z -----------------
2024-12-12T02:44:00.5472454Z 
2024-12-12T02:44:00.5472681Z Command: dpkg-buildpackage --sanitize-env -Pnocheck -us -uc -b -rfakeroot
2024-12-12T02:44:00.7050827Z dpkg-buildpackage: info: source package ros-one-libfranka
2024-12-12T02:44:00.7051646Z dpkg-buildpackage: info: source version 0.13.3-34-gfd2951b-2024.12.12.02.42
2024-12-12T02:44:00.7052326Z dpkg-buildpackage: info: source distribution jammy
2024-12-12T02:44:00.7053063Z dpkg-buildpackage: info: source changed by Franka Robotics GmbH <support@franka.de>
2024-12-12T02:44:00.7238418Z  dpkg-source --before-build .
2024-12-12T02:44:00.7238904Z dpkg-buildpackage: info: host architecture amd64
2024-12-12T02:44:00.8049631Z dpkg-source: info: using options from package/debian/source/options: --auto-commit
2024-12-12T02:44:00.8539980Z  fakeroot debian/rules clean
2024-12-12T02:44:00.8819624Z dh clean -v --buildsystem=cmake --builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:00.9657474Z    dh_auto_clean -O-v -O--buildsystem=cmake -O--builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:01.0065265Z    dh_autoreconf_clean -O-v -O--buildsystem=cmake -O--builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:01.0385715Z    dh_clean -O-v -O--buildsystem=cmake -O--builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:01.0705829Z 	rm -f debian/debhelper-build-stamp
2024-12-12T02:44:01.0706339Z 	rm -rf debian/.debhelper/
2024-12-12T02:44:01.0724483Z 	rm -f -- debian/ros-one-libfranka.substvars debian/files
2024-12-12T02:44:01.0740624Z 	rm -fr -- debian/ros-one-libfranka/ debian/tmp/
2024-12-12T02:44:01.0756935Z 	find .  \( \( \
2024-12-12T02:44:01.0757778Z 		\( -path .\*/.git -o -path .\*/.svn -o -path .\*/.bzr -o -path .\*/.hg -o -path .\*/CVS -o -path .\*/.pc -o -path .\*/_darcs \) -prune -o -type f -a \
2024-12-12T02:44:01.0758519Z 	        \( -name '#*#' -o -name '.*~' -o -name '*~' -o -name DEADJOE \
2024-12-12T02:44:01.0758937Z 		 -o -name '*.orig' -o -name '*.rej' -o -name '*.bak' \
2024-12-12T02:44:01.0759295Z 		 -o -name '.*.orig' -o -name .*.rej -o -name '.SUMS' \
2024-12-12T02:44:01.0759879Z 		 -o -name TAGS -o \( -path '*/.deps/*' -a -name '*.P' \) \
2024-12-12T02:44:01.0760219Z 		\) -exec rm -f {} + \) -o \
2024-12-12T02:44:01.0760544Z 		\( -type d -a -name autom4te.cache -prune -exec rm -rf {} + \) \)
2024-12-12T02:44:01.0846721Z  debian/rules build
2024-12-12T02:44:01.1029029Z dh build -v --buildsystem=cmake --builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:01.1798263Z    dh_update_autotools_config -O-v -O--buildsystem=cmake -O--builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:01.2140497Z    dh_autoreconf -O-v -O--buildsystem=cmake -O--builddirectory=.obj-x86_64-linux-gnu
2024-12-12T02:44:01.2458065Z    debian/rules override_dh_auto_configure
2024-12-12T02:44:01.2475630Z make[1]: Entering directory '/<<BUILDDIR>>package'
2024-12-12T02:44:01.2643369Z # In case we're installing to a non-standard location, look for a setup.sh
2024-12-12T02:44:01.2650999Z # in the install tree and source it.  It will set things like
2024-12-12T02:44:01.2658451Z # CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
2024-12-12T02:44:01.2666551Z if [ -f "/opt/ros/one/setup.sh" ]; then . "/opt/ros/one/setup.sh"; fi && \
2024-12-12T02:44:01.2667143Z dh_auto_configure -- \
2024-12-12T02:44:01.2667569Z 	-DCMAKE_INSTALL_PREFIX="/opt/ros/one" \
2024-12-12T02:44:01.2668023Z 	-DCMAKE_PREFIX_PATH="/opt/ros/one" \
2024-12-12T02:44:01.2668424Z 	-DBUILD_TESTING=OFF
2024-12-12T02:44:01.3499208Z 	install -d .obj-x86_64-linux-gnu
2024-12-12T02:44:01.3502536Z 	cd .obj-x86_64-linux-gnu && cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_RUNSTATEDIR=/run "-GUnix Makefiles" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_INSTALL_LIBDIR=lib/x86_64-linux-gnu -DCMAKE_INSTALL_PREFIX=/opt/ros/one -DCMAKE_PREFIX_PATH=/opt/ros/one -DBUILD_TESTING=OFF ..
2024-12-12T02:44:01.4599178Z -- The CXX compiler identification is GNU 11.4.0
2024-12-12T02:44:01.4700825Z -- Detecting CXX compiler ABI info
2024-12-12T02:44:01.5972689Z -- Detecting CXX compiler ABI info - done
2024-12-12T02:44:01.6053202Z -- Check for working CXX compiler: /usr/lib/ccache/c++ - skipped
2024-12-12T02:44:01.6055951Z -- Detecting CXX compile features
2024-12-12T02:44:01.6060796Z -- Detecting CXX compile features - done
2024-12-12T02:44:01.6364226Z -- Found Eigen3: /usr/include/eigen3  
2024-12-12T02:44:01.6370318Z -- pinocchio_default FOUND. pinocchio_default at /opt/ros/one/lib/x86_64-linux-gnu/libpinocchio_default.so
2024-12-12T02:44:01.6371566Z -- pinocchio_parsers FOUND. pinocchio_parsers at /opt/ros/one/lib/x86_64-linux-gnu/libpinocchio_parsers.so
2024-12-12T02:44:01.6375005Z -- boost_filesystem FOUND. boost_filesystem at /usr/lib/x86_64-linux-gnu/libboost_filesystem.so
2024-12-12T02:44:01.6377452Z -- boost_serialization FOUND. boost_serialization at /usr/lib/x86_64-linux-gnu/libboost_serialization.so
2024-12-12T02:44:01.6380083Z -- boost_system FOUND. boost_system at /usr/lib/x86_64-linux-gnu/libboost_system.so
2024-12-12T02:44:01.6764478Z -- Found Boost: /usr/include (found version "1.74.0") found components: filesystem serialization system 
2024-12-12T02:44:01.6775651Z -- eigenpy FOUND. eigenpy at /opt/ros/one/lib/x86_64-linux-gnu/libeigenpy.so
2024-12-12T02:44:02.0882393Z -- PythonLibraryDirs: 
2024-12-12T02:44:02.0882860Z -- PythonLibVersionString: 3.10.12
2024-12-12T02:44:02.1086331Z -- Python site lib: lib/python3.10/dist-packages
2024-12-12T02:44:02.1086734Z -- Python include dirs: /usr/include/python3.10
2024-12-12T02:44:02.1328585Z -- Checking for NumPy
2024-12-12T02:44:02.2543897Z --   NUMPY_INCLUDE_DIRS=/usr/lib/python3/dist-packages/numpy/core/include
2024-12-12T02:44:02.3758775Z --   NUMPY_VERSION=1.21.5
2024-12-12T02:44:02.3759382Z -- NumPy include dir: /usr/lib/python3/dist-packages/numpy/core/include
2024-12-12T02:44:02.3804620Z CMake Warning at /usr/share/cmake-3.22/Modules/FindBoost.cmake:2201 (message):
2024-12-12T02:44:02.3805454Z   No header defined for python310; skipping header check (note: header-only
2024-12-12T02:44:02.3806071Z   libraries have no designated component)
2024-12-12T02:44:02.3806515Z Call Stack (most recent call first):
2024-12-12T02:44:02.3807146Z   /opt/ros/one/lib/x86_64-linux-gnu/cmake/eigenpy/boost.cmake:144 (find_package)
2024-12-12T02:44:02.3808082Z   /opt/ros/one/lib/x86_64-linux-gnu/cmake/eigenpy/eigenpyConfig.cmake:149 (SEARCH_FOR_BOOST_PYTHON)
2024-12-12T02:44:02.3809057Z   /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
2024-12-12T02:44:02.3810150Z   /opt/ros/one/lib/x86_64-linux-gnu/cmake/pinocchio/pinocchioConfig.cmake:160 (find_dependency)
2024-12-12T02:44:02.3810609Z   CMakeLists.txt:52 (find_package)
2024-12-12T02:44:02.3810782Z 
2024-12-12T02:44:02.3810787Z 
2024-12-12T02:44:02.3817472Z -- Found Boost: /usr/include (found version "1.74.0") found components: python310 
2024-12-12T02:44:02.3821512Z -- Boost_PYTHON_LIBRARY: /usr/lib/x86_64-linux-gnu/libboost_python310.so
2024-12-12T02:44:02.7790434Z -- Default C++ standard: 201703
2024-12-12T02:44:02.7790949Z -- Minimal C++ standard upgraded to 11
2024-12-12T02:44:02.7791338Z -- C++ standard sufficient: Minimal required 11, currently defined: 17
2024-12-12T02:44:02.7804206Z -- C++ standard sufficient: Minimal required 11, currently defined: 17
2024-12-12T02:44:02.7816242Z -- Looking for C++ include pthread.h
2024-12-12T02:44:02.9205976Z -- Looking for C++ include pthread.h - found
2024-12-12T02:44:02.9207945Z -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
2024-12-12T02:44:03.0655047Z -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
2024-12-12T02:44:03.0664687Z -- Found Threads: TRUE  
2024-12-12T02:44:03.0689035Z -- fmt not found. Fetching fmt...
2024-12-12T02:44:03.0966152Z CMake Error at /usr/share/cmake-3.22/Modules/ExternalProject.cmake:2666 (message):
2024-12-12T02:44:03.0966947Z   error: could not find git for clone of fmt-populate
2024-12-12T02:44:03.0967499Z Call Stack (most recent call first):
2024-12-12T02:44:03.0968159Z   /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3716 (_ep_add_download_command)
2024-12-12T02:44:03.0968859Z   CMakeLists.txt:23 (ExternalProject_Add)
2024-12-12T02:44:03.0969164Z 
2024-12-12T02:44:03.0969189Z 
2024-12-12T02:44:03.0969386Z -- Configuring incomplete, errors occurred!
2024-12-12T02:44:03.0970337Z See also "/<<BUILDDIR>>package/.obj-x86_64-linux-gnu/_deps/fmt-subbuild/CMakeFiles/CMakeOutput.log".
2024-12-12T02:44:03.0970916Z 
2024-12-12T02:44:03.0971256Z CMake Error at /usr/share/cmake-3.22/Modules/FetchContent.cmake:1075 (message):
2024-12-12T02:44:03.0971875Z   CMake step for fmt failed: 1
2024-12-12T02:44:03.0972266Z Call Stack (most recent call first):
2024-12-12T02:44:03.0972970Z   /usr/share/cmake-3.22/Modules/FetchContent.cmake:1216:EVAL:2 (__FetchContent_directPopulate)
2024-12-12T02:44:03.0973778Z   /usr/share/cmake-3.22/Modules/FetchContent.cmake:1216 (cmake_language)
2024-12-12T02:44:03.0974175Z   cmake/FetchFMT.cmake:13 (FetchContent_Populate)
2024-12-12T02:44:03.0974461Z   CMakeLists.txt:76 (include)
2024-12-12T02:44:03.0974620Z 
2024-12-12T02:44:03.0974623Z 
2024-12-12T02:44:03.0974729Z -- Configuring incomplete, errors occurred!
2024-12-12T02:44:03.0975122Z See also "/<<BUILDDIR>>package/.obj-x86_64-linux-gnu/CMakeFiles/CMakeOutput.log".
2024-12-12T02:44:03.1016184Z 	cd .obj-x86_64-linux-gnu && tail -v -n \+0 CMakeCache.txt
2024-12-12T02:44:03.1031497Z ==> CMakeCache.txt <==
2024-12-12T02:44:03.1033437Z # This is the CMakeCache file.
2024-12-12T02:44:03.1034243Z # For build in directory: /<<BUILDDIR>>package/.obj-x86_64-linux-gnu
2024-12-12T02:44:03.1034987Z # It was generated by CMake: /usr/bin/cmake
2024-12-12T02:44:03.1035543Z # You can edit this file to change values found and used by cmake.
2024-12-12T02:44:03.1036228Z # If you do not want to change any of the values, simply exit the editor.
2024-12-12T02:44:03.1036910Z # If you do want to change a value, simply edit, save, and exit the editor.
2024-12-12T02:44:03.1037483Z # The syntax for the file is as follows:
2024-12-12T02:44:03.1037889Z # KEY:TYPE=VALUE
2024-12-12T02:44:03.1038237Z # KEY is the name of a variable in the cache.
2024-12-12T02:44:03.1038778Z # TYPE is a hint to GUIs for the type of VALUE, DO N
```

AndreasKuhner commented:
Hi @k-okada ,
thank you for the PR. Regarding the second change to move fmt from private to public -> The user has the possibility to leverage our logging capabilities. Hence, would need also the FMT dependency under `PUBLIC`. The package.xml addition would be something for being merged :)

Cheers,
Andreas

AndreasKuhner commented:
That should be a missing `find_package` call for fmt aka `find_package(fmt REQUIRED)`. You can try to add it in the franka_hw `CMakeLists.txt`

Theoretically, this should come via the linking against `libfranka` :thinking: . By any chance, maybe you can also check if you have installed `fmt`. But that should also usually come along ROS stuff - it is used heavily there. (aka `sudo apt install libfmt-dev`).

k-okada commented:
@AndreasKuhner thank you, i have removed PRIVATE/PUBLIC commit from this PR and added `find_package(fmt REQUIRED) to https://github.com/rhaschke/franka_ros/pull/4

cc: @rhaschke

rhaschke commented:
> That should be a missing `find_package` call for fmt aka `find_package(fmt REQUIRED)`.  Theoretically, this should come via the linking against `libfranka`.

Perfectly correct. For this reason, the issue should be fixed here, not in `franka_ros`. You are missing the transitive dependency declaration. I filed a corresponding PR: #173.

AndreasKuhner commented:
Was fixed with another PR -> If anything is still open, please let us know.
